pkg/scheduler/actions/allocate/allocate.go

```diff
@@ -250,4 +254,32 @@ func (alloc *Action) Execute(ssn *framework.Session) {
 	}
 }
 
+func prePredicateforAllocate(ssn *framework.Session, task *api.TaskInfo, allNodes []*api.NodeInfo, job *api.JobInfo) error {
```

Comment:
The `prePredicateforAllolcate` is similar with `prePredicateforBackfill` and `prePredicateforReclaim`

Reply:
done